internal_packages/keybase/lib/keybase-user.cjsx

```diff
@@ -0,0 +1,237 @@
+    )
+    return
+
+  _delete: (event) =>
+    # delete a locally saved key
+    email = event.target.attributes['data-email'].value
+    keys = PGPKeyStore.pubKeys(email)
+    if keys? and keys.length == 1
+      key = keys[0]
+    else if keys.length > 1
+      # TODO key metadata objects need a unique ID
+      key = @_matchKeys(@props.profile.key, keys)
+      #console.error "Multiple keys associated with #{email}, unable to determine which to remove from"
+
+    if key?
+      PGPKeyStore.deleteKey(key)
+    else
+      console.error "Unable to fetch key for #{email}"
+
+  _addEmail: (email) =>
+    # associate another email address with this key
+    PGPKeyStore.addAddressToKey(@props.profile, email)
+
+  _addEmailInput: (contacts) =>
+    # when a new email is added to the list of emails
+    # flow is (click "add email") -> _addEmailClick -> (enter email) -> this
+    emails = _.pluck(contacts.to, 'email')
+    _.each(emails, @_addEmail)
+    @setState({inputEmail: false})
+
+  _addEmailClick: (event) =>
+    # create a text field in which to enter a new email to associate with a key
+    @setState({inputEmail: true})
+    # TODO focus on the new field
+    # React.findDOMNode(@refs.addNewEmail).focus()
+
+  _removeEmailClick: (event) =>
+    email = event.target.attributes['data-email'].value
+    @_removeEmail(email)
+
+  _removeEmail: (email) =>
+    PGPKeyStore.removeAddressFromKey(@props.profile, email)
+
+  render: =>
+    {profile} = @props
+
+    keybaseDetails = null
+    if profile.keybase_user?
+      keybase = profile.keybase_user
+      # username
+      username = keybase.components.username.val
+
+      # full name
+      if keybase.components.full_name?.val?
+        fullname = keybase.components.full_name.val
+      else
+        fullname = username
+
+      # profile picture
+      if keybase.thumbnail?
+        picture = keybase.thumbnail
+      else
+        picture = "#"
+        # TODO default picture
+
+      # various web accounts/profiles
+      possible_profiles = ["twitter", "github", "coinbase"]
+      profiles = _.map(possible_profiles, (possible) =>
+        if keybase.components[possible]?.val?
+          return "#{possible}: #{keybase.components[possible].val}"
+      )
+      profiles = _.reject(profiles, (profile) -> profile is undefined)
+
+      if profiles.length > 0
+        profiles =  _.map(profiles, (profile) ->
+          return <li>{ profile }</li>)
+        profileList = (<ul>{ profiles }</ul>)
+      else
+        profileList = null
+
+      keybaseDetails = (<div className="details">
+        <h2>
+          { fullname }&nbsp;
+          <small className="profile-username">
+            ({ username })
+          </small>
+        </h2>
+
+        { profileList }
+      </div>)
+
+    # button to save/delete key
+    if profile.addresses?.length > 0
+      # TODO what about the other email(s) that might be in the list?
+      saved = PGPKeyStore.pubKeys(profile.addresses[0])
```

Comment:
We should technically call this method when the react component gets it's `@props.profile` and save it to state.